pymatgen/analysis/tests/test_phase_diagram.py

```diff
@@ -90,30 +91,37 @@ def test_str(self):
     def test_read_csv(self):
         entries = EntrySet.from_csv(str(module_dir / "pdentries_test.csv"))
         self.assertEqual(entries.chemsys, {"Li", "Fe", "O"}, "Wrong elements!")
-        self.assertEqual(len(entries), 492, "Wrong number of entries!")
+        self.assertEqual(len(entries), 490, "Wrong number of entries!")
 
 
 class TransformedPDEntryTest(unittest.TestCase):
     def setUp(self):
+        # NOTE a 1:1 isn't the best test? maybe a tertiary system should be used.
         comp = Composition("LiFeO2")
         entry = PDEntry(comp, 53)
-        self.transformed_entry = TransformedPDEntry(
-            {DummySpecies("Xa"): 1, DummySpecies("Xb"): 1}, entry
-        )
+
+        terminal_compositions = ["LiO", "FeO"]
+        terminal_compositions = [Composition(c) for c in terminal_compositions]
+
+        sp_mapping = OrderedDict()
+        for i, comp in enumerate(terminal_compositions):
+            sp_mapping[comp] = DummySpecies("X" + chr(102 + i))
+
+        self.transformed_entry = TransformedPDEntry(entry, sp_mapping)
 
     def test_get_energy(self):
         self.assertEqual(self.transformed_entry.energy, 53, "Wrong energy!")
-        self.assertEqual(self.transformed_entry.original_entry.energy, 53.0)
+        self.assertAlmostEqual(self.transformed_entry.original_entry.energy, 53.0, 11)
 
     def test_get_energy_per_atom(self):
-        self.assertEqual(self.transformed_entry.energy_per_atom, 53.0 / 2)
+        self.assertAlmostEqual(self.transformed_entry.energy_per_atom, 53.0 / 2, 11)
 
     def test_get_name(self):
         self.assertEqual(self.transformed_entry.name, "LiFeO2", "Wrong name!")
 
     def test_get_composition(self):
         comp = self.transformed_entry.composition
-        expected_comp = Composition({DummySpecies("Xa"): 1, DummySpecies("Xb"): 1})
+        expected_comp = Composition({DummySpecies("Xf"): 1, DummySpecies("Xg"): 1})
```

Comment:
Is there any reason the DummySpecies changed here? (Not a problem, just wanting to make sure I hadn't missed something important)

Reply:
just to mimic the dictionary you would get for this task using the live code